docs/Formula-Cookbook.md

```diff
@@ -1052,6 +1052,24 @@ The `sockets` method accepts a formatted socket definition as `<type>://<host>:<
 
 Please note that sockets will be accessible on IPv4 and IPv6 addresses by default.
 
+If you only need one socket and you don't care about the name (the default is `Listeners`):
+
+```rb
+service do
+  run [opt_bin/"beanstalkd", "test"]
+  sockets "tcp://127.0.0.1:80"
+end
+```
+
+If you need multiple sockets and/or you want to specify the name:
+
+```rb
+service do
+  run [opt_bin/"beanstalkd", "test"]
+  sockets "Socket" => "tcp://0.0.0.0:80", "SocketTLS" => "tcp://0.0.0.0:443"
```

Comment:
```suggestion
  sockets socket: "tcp://0.0.0.0:80", socket_tls: "tcp://0.0.0.0:443"
```

or something might be a nicer/more Ruby-like API. Not huge on the `socket:`/`socket_tls:` naming, though; would welcome better ones.

Reply:
I think using a string here gives users more flexibility when it comes to matching the exact names of the sockets they already use internally. Or at least that was my thinking.

Reply:
> I think using a string here gives users more flexibility when it comes to matching the exact names of the sockets they already use internally.

@apainintheneck Can you elaborate on this a bit? Do the names of these sockets matter and, if so, when/how are they used? 

To be clear: this may be entirely just correctly my understanding 😁

Reply:
I didn't know what I was talking about 😄.

From the man page `man launchd.plist`:

```
Sockets <dictionary of dictionaries... OR dictionary of array of dictionaries...>
This optional key is used to specify launch on demand sockets that can be used to let launchd know when to run the job. The job must check-in to get a copy of the file descriptors using the launch_activate_socket(3) API.  The keys of the top level Sockets dictionary can be anything. These keys are meant for the application developer to associate which socket descriptors correspond to which application level protocols (e.g. http vs. ftp vs. DNS...).
```

Looking at it again we use symbols for rest of the service DSL. I'll change the code to use symbols.